// 4devs/dist/modules/Certificates/GenerateCertificates.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.GenerateCertificates = void 0;
const RequestUtil_1 = require("../../utils/RequestUtil");
class GenerateCertificates {
    constructor() {
        this.requestUtil = new RequestUtil_1.RequestUtil();
    }
    async execute({ isWithDots, type, }) {
        const typeParamToRequest = await this.handleTypeToRequest(type);
        const { data: certificate } = await this.requestUtil.post({
            path: '/ferramentas_online.php',
            json: {
                acao: 'gerador_certidao',
                pontuacao: isWithDots ? 'S' : 'N',
                tipo_certidao: typeParamToRequest,
            },
        });
        return {
            certificate: certificate,
        };
    }
    async handleTypeToRequest(type) {
        switch (type) {
            case 'BIRTH':
                return 'nascimento';
            case 'WEDDING':
                return 'casamento';
            case 'RELIGIOUS_MARRIAGE':
                return 'casamento_religioso';
            case 'DEATH':
                return 'obito';
            default:
                return 'Indiferente';
        }
    }
}
exports.GenerateCertificates = GenerateCertificates;
